pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -145,6 +145,15 @@ func (r *ReconcileComplianceRemediation) Reconcile(request reconcile.Request) (r
 		reconcileErr = r.reconcileRemediation(remediationInstance, reqLogger)
 	}
 
+	if remediationInstance.HasNotSetValues() {
+		res, valueErr := r.handleUnsetValues(remediationInstance, reqLogger)
+		if res.Requeue || valueErr != nil {
+			return res, valueErr
+		}
+	} else {
+		reconcileErr = r.reconcileRemediation(remediationInstance, reqLogger)
+	}
+
```

Comment:
Hi Jakub, I have changed to this `if remediationInstance.Spec.Apply && remediationInstance.HasUnmetDependencies() {
		reqLogger.Info("Has unmet dependencies. Requeuing")
		return r.handleUnmetDependencies(remediationInstance, reqLogger)

	}

	if remediationInstance.Spec.Apply && remediationInstance.HasUnsetValues() {
		reqLogger.Info("Has unset values. Requeuing")
		return r.handleUnsetValues(remediationInstance, reqLogger)
	}`

Does this look ok to you? Thanks

Reply:
Maybe the context around this (old) comment has changed, but why don't we use:
```
if remediationInstance.HasUnmetDependencies() {
 ...
} else if remediationInstance.HasAnnotation(compv1alpha1.RemediationUnsetValueAnnotation) {
 ...
} else if remediationInstance.HasAnnotation(compv1alpha1.RemediationValueRequiredAnnotation) {
 ...
} else {
 reconcileErr = r.reconcileRemediation(remediationInstance, reqLogger)
}
```
?

Reply:
Because there are situations where we have both unset-values annotation and value required Annotation,  and since the second will always be true, it will never reach the third block.